packages/material-ui/src/useMediaQuery/useMediaQuery.js

```diff
@@ -1,37 +1,53 @@
 import React from 'react';
+import warning from 'warning';
+import { getThemeProps, useTheme } from '@material-ui/styles';
 
 // This variable will be true once the server-side hydration is completed.
 let hydrationCompleted = false;
 
-function deepEqual(a, b) {
-  return a.length === b.length && a.every((item, index) => item === b[index]);
-}
-
 function useMediaQuery(queryInput, options = {}) {
-  const multiple = Array.isArray(queryInput);
-  let queries = multiple ? queryInput : [queryInput];
-  queries = queries.map(query => query.replace('@media ', ''));
+  const theme = useTheme();
+  const props = getThemeProps({
+    theme,
+    name: 'MuiUseMediaQuery',
+    props: {},
+  });
+
+  warning(
+    typeof queryInput !== 'function' || theme !== null,
+    [
+      'Material-UI: the `query` argument provided is invalid.',
+      'You are providing a function without a theme in the context.',
+      'One of the parent elements needs to use a ThemeProvider.',
+    ].join('\n'),
+  );
+
+  let query = typeof queryInput === 'function' ? queryInput(theme) : queryInput;
+  query = query.replace('@media ', '');
```

Comment:
```suggestion
  query = query.replace(/^@media( ?)/m, '');
```
This will let both `@media(...)` and `@media (...)` work

Reply:
So far the objective of this logic is to handle the `theme.breakpoints` helpers. Do we need to extend the support? I have removed the usage of @media from the demos.

Reply:
Ok, a developer could have its own breakpoint helpers. Given that CSS support the "without space" case. I think that we should too. Thanks for rasing it 👍

Reply:
Could you commit the change?